src/network/websockets/server/index.ts

```diff
@@ -50,20 +50,27 @@ export function runWebsocketServer(params: Params) {
     const handlers: { [action: string]: (msg: any) => void } = {
       ping: () => send({ action: "pong" }),
       auth: msg => {
-        const { token } = msg;
-        onAuthenticate({ token })
-          .then(function(result) {
-            send({
-              action: "authResult",
-              result: result
-            });
-          })
-          .catch(function() {
-            send({
-              action: "authResult",
-              result: "internalError"
-            });
+        if (!onAuthenticate) {
+          send({
```

Comment:
Perhaps we should even throw? If we don’t support authentication and the client sends us authentication, I would rather let the developer know that they have a bug in the code.

```ts
// Before
if (!onAuthenticate) {
  send({
    action: "authResult",
    result: "Not Implemented"
  });
}

// After
if (!onAuthenticate)
  throw new Error("Unexpected auth message");
```